stats/statsreporter.go

```diff
@@ -88,24 +91,32 @@ func Post(destination string, stats []Sample) error {
 	payload := map[string][]Sample{"metrics": stats}
 	data, err := json.Marshal(payload)
 	if err != nil {
-		plog.Warningf("Couldn't marshal stats: ", err)
+		plog.WithField("stats", stats).WithError(err).
+			Warn("Couldn't marshal stats")
```

Comment:
This should /maybe/ be debug, but warning is ok. Nonetheless it needs to be more human readable.